lightning/mydump/bytes.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package bytes implements functions for the manipulation of byte slices.
+// It is analogous to the facilities of the strings package.
+
+// this is copy from `bytes/bytes.go`
+
+package mydump
+
+import "unicode/utf8"
+
+// asciiSet is a 32-byte value, where each bit represents the presence of a
+// given ASCII character in the set. The 128-bits of the lower 16 bytes,
+// starting with the least-significant bit of the lowest word to the
+// most-significant bit of the highest word, map to the full range of all
+// 128 ASCII characters. The 128-bits of the upper 16 bytes will be zeroed,
+// ensuring that any non-ASCII character will be reported as not in the set.
+type asciiSet [8]uint32
+
+// makeASCIISet creates a set of ASCII characters and reports whether all
+// characters in chars are ASCII.
+func makeASCIISet(chars string) (as asciiSet, ok bool) {
+    for i := 0; i < len(chars); i++ {
+        c := chars[i]
+        if c >= utf8.RuneSelf {
+            return as, false
+        }
+        as[c>>5] |= 1 << uint(c&31)
+    }
+    return as, true
+}
+
+// contains reports whether c is inside the set.
+func (as *asciiSet) contains(c byte) bool {
+    return (as[c>>5] & (1 << uint(c&31))) != 0
+}
+
+// IndexAnyAscii returns the byte index of the first occurrence in s of any of the Unicode
+// code points in chars. It returns -1 if  there is no code
+// point in common.
+func IndexAnyAscii(s []byte, as *asciiSet) int {
+    for i, c := range s {
+        if as.contains(c) {
+            return i
+        }
+    }
+    return -1
+
+}
+
+func IndexAnyUtf8(s []byte, chars string) int {
```

Comment:
we don't need to care about UTF-8. the current CSV restriction is that the separator must be one *byte* long and the delimiter must be zero or one *byte* long, which leaves no room for non-ASCII characters.

Reply:
The implement is consistent with both  ascii and utf-8 string and with automatically use either `IndexAnyAscii` or `IndexAnyUtf8`, so we can just left this function so it can be used if later we support utf8 delimiters

Reply:
the result will be wrong if the buffer is cut like:

```
[20 20 20 20 20 20 20 20 C3]
[A4 20 20 20 20 20 20 20 20]
```

if we search for `C3 A4` (ä) we will not found anything using `IndexAnyUtf8`.

Reply:
yes, based on current implement, it's not easy to handle this, so I removed `IndexAnyUtf8` related code